lib/internal/streams/writable.js

```diff
@@ -259,11 +258,17 @@ function WritableState(options, stream, isDuplex) {
 function resetBuffer(state) {
   state.buffered = [];
   state.bufferedIndex = 0;
-  state.state |= kAllBuffers | kAllNoop;
+  state.state |= kAllBuffers | kSimpleBuffer;
 }
 
 WritableState.prototype.getBuffer = function getBuffer() {
-  return ArrayPrototypeSlice(this.buffered, this.bufferedIndex);
+  return (this.state & kSimpleBuffer) === 0
+    ? this.buffered.slice(this.bufferedIndex)
```

Comment:
```suggestion
    ? ArrayPrototypeSlice, this.buffered, this.bufferedIndex)
```

And ArrayPrototypeMap below probably